packages/flutter/lib/src/widgets/asset_vendor.dart

```diff
@@ -0,0 +1,57 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/services.dart';
+
+import 'basic.dart';
+import 'framework.dart';
+
+/// Establishes an asset resolution strategy for its descendants.
+///
+/// An AssetVendor establishes a [ResolvingAssetBundle] as the default bundle
+/// for its subtree. The resolution logic is provided by an [AssetResolver]
+/// specified via its [AssetResolverBuilder].
+class AssetVendor extends StatefulComponent {
```

Comment:
OK.  I was somewhat following a discussion earlier in the design cycle with @Hixie where it seemed like the right thing to do was to create the system in layers, but I can simplify.